### PR TITLE
cypress: test for copy/paste bold formatted lists

### PR DESCRIPTION
**Pull request by nileshgulia1**
These covers bold lists formatted as slate content. This should be generalised to work with external copied bold content as well.

### CODE REVIEW

**Comment by nileshgulia1** [comment]
@silviubogan: Here are a few issues with the copy/paste in cypress: 
1. We can't copy (cmd+c) and paste(cmd+v) , these just don't work.
2. Manually altering clipboard contents can be another approach. These would require [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem#writing_to_clipboard)

And then another problem occurs when you append DOM content in slate which works in static HTML (in Edit mode) but when you save the page, the appended content disappears.
It looks like we need to alter the slate's data(props) somehow to make copy/paste work in cypress. https://github.com/abhinaba-ghosh/cypress-react-selector/issues/23
idk if we really need to manipulate clipboard contents. So, I ended up intercepting `GET_CONTENT` action and stub the response to get the desired data as props in slate.
 -cc @tiberiuichim

File: cypress/integration/block-slate-format-boldlists.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import { slateBeforeEach, slateAfterEach } from '../support';
```
Comment (nileshgulia1) [inline cypress/integration/block-slate-format-boldlists.js:1]: I need to rename the file.

File: src/editor/SlateEditor.jsx [hunk 2]
```diff
@@ -290,6 +290,12 @@ class SlateEditor extends Component {
                 if (handled) return;
                 onKeyDown && onKeyDown({ editor, event });
               }}
+              onPaste={(event) => {
```
Comment (tiberiuichim) [inline src/editor/SlateEditor.jsx:293]: I think it's safer to have it here as:
```
onPaste={this.props.onPaste ? (event) => {...} : undefined}
```

File: src/editor/SlateEditor.jsx [hunk 3]
```diff
@@ -290,6 +290,12 @@ class SlateEditor extends Component {
...
+                if (this.props.onPaste) this.props.onPaste({ editor, event });
+              }}
+              onCopy={(event) => {
+                if (this.props.onCopy) this.props.onCopy({ editor, event });
```
Comment (tiberiuichim) [inline src/editor/SlateEditor.jsx:297]: same with onCopy

**Review by tiberiuichim — APPROVED** [review]
Awesome! Cypress tests are starting to look good.

A bit of cleanup needed, there's some code commented.

**Comment by silviubogan** [comment]
@nileshgulia1 I think that besides the removal of the useless comments, if the Cypress tests pass on a Linux installation too, this PR is good to be merged. Thanks.

File: cypress/integration/block-slate-format-boldlists.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+  it('log browser info', () => {
+    Cypress.log(Cypress.browser);
+  });
+
+  it('gives clipboard permission to browser ', () => {
+    // use the Chrome debugger protocol to grant the current browser window
+    // access to the clipboard from the current origin
+    cy.wrap(
+      Cypress.automation('remote:debugger:protocol', {
```
Comment (nileshgulia1) [inline cypress/integration/block-slate-format-boldlists.js:15]: https://github.com/cypress-io/cypress/issues/7942

**Comment by nileshgulia1** [comment]
@silviubogan This is ready now!

**Review by avoinea — APPROVED** [review]

**Pull request merged**